vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -944,6 +945,13 @@ window.Vaadin.Flow.gridConnector = {
     function _fireClickEvent(event, eventName) {
       if (grid.$connector.activeItem) {
         event.itemKey = grid.$connector.activeItem.key;
+        const eventContext = grid.getEventContext(event);
+        if (eventContext.column.id) {
+          event.columnKey = eventContext.column.id;
+        }
+        // flowId should be always set and will be used as
+        // backup when the user forgets about setting column's id
+        event.flowKey = eventContext.column._flowId;
```

Comment:
If you have a details-renderer, and click the details row, it should fire an `ItemClickEvent`, but `getEventContext().column` is undefined. So this needs a falsy-check.
This should also have a test-case.

Reply:
in this case, is it possible to obtain the clicked column some other way?

Reply:
No, because the click doesn't target any column in the case of details cell, which spans through the whole grid. That's why `getEventContext` has been designed to not have a column-reference in this case.

Reply:
It still needs a test case (clicking a details cell).

Reply:
what would be the correct thing to do, because all have pros and cons:
1. build a separate IT test for that
2. check it in e.g. `GridDetailsRowIT`
3. modify existing `ItemClickListenerIT` to provide details for the row

Reply:
3. would be most appropriate IMO